database.js

```diff
@@ -0,0 +1,99 @@
+class Database
+{
+    constructor()
+{
+    this._teamList = [];
+    this.tempId = [];
+    this.id = 0;
+}
+getTeams()
+{
+    return this._teamList;
+}
+getNewId()
+{
+    this.tempId = [];
+    this._teamList.forEach(team =>
+    {
+        this.tempId.push(team.id);
+        this.tempId.sort((a, b) => b - a);
+        this.id = this.tempId[0];
+    })
+    if (this.tempId.length >0)
+    { return (this.tempId[0]+1);}
+    else 
+    { return 1; }
+}
+updateLocalTeamList(teamList)
+{
+    teamList = JSON.parse(teamList)
+    this._teamList = [];
+    teamList.Items.forEach(team =>
+    {
+        this._teamList.push({id : parseInt(team.id), name : team.TeamName});
+    })
+    return this._teamList;
+}
+removeTeam(team)
+{
+    function findTeam(element)
+    {
+        return element.name == team;
+    }
+    var idxTeam = this._teamList.findIndex(findTeam)
+    var Promise1 = new Promise((resolve, reject) => 
+    {   
+        const xhr = new XMLHttpRequest();
+        xhr.open("DELETE", apiAdress);
+        xhr.setRequestHeader("x-api-key", apiKey);
+        xhr.setRequestHeader("Content-Type", "application/json");
+        xhr.onload = () => resolve(xhr.responseText);
+        xhr.onerror = () => reject(xhr.statusText);
+        var jsonResult = 
+        {
+        "TableName": "foo", 
+        "Key": {"id": this._teamList[idxTeam].id.toString()}
+        }
+        xhr.send(JSON.stringify(jsonResult));
+        this.getTeamsFromDb();
```

Comment:
a co to jest?? Odpalasz async request o usunięcie drużyny, w miedzczasie uruchamiasz drugi async request, kompletnie nieskorelowany z tamtym i masz nadzieję że wszystko się zgra? :)

Reply:
Co więcej, ten drugi, który pobiera bieżące drużyny leci sobie w gdzieś sam w kosmosie i nawet ie można na niego poczekać na zewnąrz żeby wszystko się razem wykonało

Reply:
Poprawione.